### ISSUE TITLE
[docdb] Improve load balancer detection of when to start global load balancing

### ISSUE THREAD

**Issue by hulien22** [yugabyte/yugabyte-db#5868]
Global load balancing was recently added in #3079 (commit 83bf39c). In that initial implementation, we wait for a run of the load balancer to have no tasks/errors to determine if we are done table balancing and are ready to begin global balancing:
https://github.com/yugabyte/yugabyte-db/blob/68430a389f4326e474173b639ae6a47af4564939/src/yb/master/cluster_balance.cc#L410

However, `ai.isIdle()` includes leader stepdown tasks which are not necessary to wait for in order to start global load balancing. Thus, we can improve here by just waiting until all table moves are done. The slightly tricky thing here is that a move operation involves an add task followed by a remove task on a later run, and we have no way of linking these two tasks to each other - one potential improvement then is to wait for no add/remove tasks in the last run instead.

**Comment by bmatican** [comment]
One extra note there @hulien22 is that, we might not issue another add, due to 2 constraints
- we have a cap on outstanding adds
- the previous adds are not done (eg: tablets are still being remote or locally bootstrapped, as they are pretty large)

Given that, we could easily be in a situation where, for 1 run we schedule an add, then it takes minutes for it to finish, so for many runs we do not schedule any new adds (and presumably no removes either). Is that something that's already being covered?

**Comment by hulien22** [comment]
@bmatican Yes that situation is currently being covered by the fact that when we have hit the cap on outstanding adds and try to perform another add, this new add will fail and result in a `master_error`: https://github.com/yugabyte/yugabyte-db/blob/68430a389f4326e474173b639ae6a47af4564939/src/yb/master/cluster_balance.cc#L321-L323
`ai.IsIdle()` is false when `master_errors` is nonzero, so we would continue to wait to perform global moves.
So for future improvements, we should probably also look at `master_errors` and/or the pending add/remove tasks.
